server/src/main/java/org/elasticsearch/index/fieldvisitor/SingleFieldsVisitor.java

```diff
@@ -46,7 +46,7 @@ public Status needsField(FieldInfo fieldInfo) {
     }
 
     private void addValue(Object value) {
-        destination.add(field.valueForDisplay(value));
+        destination.add(value);
```

Comment:
Just for my curiosity, Was this call (#valueForDisplay) not necessary?

Reply:
It's moved into FieldLookup.setValues(), mainly because there are now two paths that can set it (here in the stored fields provider and also in the fetchphase pre-loaded implementation)